src/table/selection/selection-control.tsx

```diff
@@ -55,7 +54,7 @@ export function SelectionControl({
   // native checkboxes do not have focus move via keyboard, we implement it here programmatically
   const handleKeyDown = (event: KeyboardEvent) => {
     setShiftState(event);
-    if (isMultiSelection && tableRole !== 'grid') {
```

Comment:
This check does not work for tables with expandable rows because those have tableRole="treegrid". Besides, the role is set even if the grid navigation is turned off.